.github/workflows/asan.yml

```diff
@@ -0,0 +1,58 @@
+name: asan
+on:
+  schedule:
+    - cron: '0 0 * * 6'
+
+jobs:
+  build:
+    name: asan
+    runs-on: ubuntu-18.04
+    strategy:
+        fail-fast: false
+    steps:
+        - name: Cleanup pre-installed tools
+          run: |
+            # This is a fix for https://github.com/actions/virtual-environments/issues/1918
+            sudo rm -rf /usr/share/dotnet
+            sudo rm -rf /opt/ghc
+            sudo rm -rf "/usr/local/share/boost"
+            sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+        - name: Checkout
+          uses: actions/checkout@v2
+        - name: Create artifact directory
+          run: mkdir -p ${{ github.workspace }}/artifact
+        #- name: Configure core dump location
+        #  run: |
+            # Uncomment the line below to enable core dump collection
+            # echo "/concord-bft/build/cores/core.%e.%p" | sudo tee /proc/sys/kernel/core_pattern
+            # Uncomment this is you want to login into the running session.
+            # Please note that the build will block on this step.
+            # Refer to https://github.com/marketplace/actions/debugging-with-tmate
+        #- name: Setup tmate session
+        #  uses: mxschmitt/action-tmate@v2
+        - name: Build and test
+          run: |
+              script -q -e -c "make pull"
+              sudo df -h
+              script -q -e -c "CONCORD_BFT_CMAKE_LEAKCHECK=TRUE make build" \
+              && script -q -e -c "make test"
+        - name: Check if ASAN passed
+          if: always()
+          run: |
+              chmod +x "./.github/fail_action_if_sanitizer_reports_exist.sh"
```

Comment:
Just curiouis - why we need this?
Committing the file with +x doesn't help?

Reply:
> Just curiouis - why we need this?
> Committing the file with +x doesn't help?

To be honest, I took it as-is from the single source of truth aka Stack Overflow :)
If you think it is too lame I can change it.

Reply:
No, pure curiosity. It's not a big deal.
Besides who we are to question StackOverflow! :)